id_minter/src/main/scala/uk/ac/wellcome/platform/idminter/utils/Identifiable.scala

```diff
@@ -4,12 +4,24 @@ import scala.util.Random
 
 object Identifiable {
   private val identifierLength = 8
+
   private val forbiddenLetters = List('o', 'i', 'l', '1')
-  private val letterRange = ('1' to '9') ++ ('a' to 'z')
-  private val letterSet = letterRange.filterNot(forbiddenLetters.contains)
+  private val numberRange = ('1' to '9')
+  private val letterRange = ('a' to 'z')
+
+  private val characterSet = numberRange ++ letterRange
+  private val allowedCharacterSet = (characterSet).filterNot(forbiddenLetters.contains)
+  private val firstCharacterSet = allowedCharacterSet.filterNot(numberRange.contains)
 
-  def generate =
+  def generate = {
     (1 to identifierLength)
-      .map(_ => letterSet(Random.nextInt(letterSet.length)))
+      .map(x => x match {
+        // In certain contexts, IDs are used as a string.  This can cause
```

Comment:
It doesn't really have to do with IDs as strings, but as IDs used in XML. As one of the serialization formats of RDF is XML, it has to comply to XML rules. Apparently rules for XML IDs are that they cannot start with numbers: http://stackoverflow.com/questions/1077084/what-characters-are-allowed-in-dom-ids . As an example, Jena - one of the RDF frameworks - just assumes that everything in the model follows this rule, so it ignores numbers at the beginning of ids (crappy behavior, I know)